src/index.rs

```diff
@@ -49,6 +51,14 @@ pub struct IndexEntry {
     #[serde(default)]
     pub mtime: i64,
 
+    /// Discretionary Access Control permissions (such as read/write/execute on unix)
+    #[serde(default)]
+    pub umode: UnixMode,
```

Comment:
I like the style rule of making the variable name match the type name, so could you please `s/umode/unix_mode/`? (Also on the function names and so on.)

Reply:
Done